### PR TITLE
Set CMAKE_OSX_DEPLOYMENT_TARGET for all Mono AOT compilers on macOS

### LINKED ISSUE #57431: Blazor WASM AOT not working on MacOS 10.13.6
Installed 6.0-preview-7 SDK on MacOS 10.13.6 (High Sierra)

Ran "sudo dotnet workload install wasm-tools", installed OK

dotnet new blazorwasm, add "RunAOTCompilation" to csproj.

dotnet publish -c Release generates the following error:

/usr/local/share/dotnet/packs/Microsoft.NET.Runtime.WebAssembly.Sdk/6.0.0-preview.7.21377.19/Sdk/WasmApp.Native.targets(443,5): warning : [/Users/adam/Repos/blazorwasm60pre7/obj/Release/net6.0/linked/Microsoft.AspNetCore.Components.dll] dyld: lazy symbol binding failed: Symbol not found: ____chkstk_darwin [/Users/adam/Repos/blazorwasm60pre7/blazorwasm60pre7.csproj]

Some Googling leads me to suspect it may have something to do with the workload package itself being built on MacOS 10.15

### PR DESCRIPTION
**Pull request by akoeplinger**
Not just if we're targetting iOS/tvOS/macOS since the AOT compiler for WASM or Android can run on macOS too.

Fixes https://github.com/dotnet/runtime/issues/57431

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -546,24 +546,32 @@
...
+    </ItemGroup>
     <PropertyGroup Condition="'$(TargetOS)' == 'Linux' and '$(Platform)' == 'arm64'">
       <MonoUseCrossTool>true</MonoUseCrossTool>
       <MonoAotAbi>aarch64-linux-gnu</MonoAotAbi>
       <MonoAotOffsetsFile>$(MonoObjCrossDir)offsets-aarch-linux-gnu.h</MonoAotOffsetsFile>
       <MonoAotOffsetsPrefix>$(MonoCrossDir)/usr/lib/gcc/aarch64-linux-gnu/5</MonoAotOffsetsPrefix>
     </PropertyGroup>
 
+    <!-- macOS specific options -->
+    <ItemGroup Condition="$([MSBuild]::IsOSPlatform('OSX'))">
```
Comment (lambdageek) [inline src/mono/mono.proj:571]: nit: "macOS host specific options" - a lot of the other stuff here is about the cross compiler target.

Isn't `[MSBuild]::IsOSPlatform` checking the build machine? not the host?  I think you want to compare `RealTargetOS` if it is set.

File: src/mono/mono.proj [hunk 2]
```diff
@@ -546,24 +546,32 @@
...
       <MonoAotOffsetsFile>$(MonoObjCrossDir)offsets-$(Platform)-darwin.h</MonoAotOffsetsFile>
       <MonoAotAbi Condition="'$(Platform)' == 'arm64'">aarch64-apple-maccatalyst</MonoAotAbi>
       <MonoAotAbi Condition="'$(Platform)' == 'x64'">x86_64-apple-maccatalyst</MonoAotAbi>
     </PropertyGroup>
-    <ItemGroup Condition="'$(TargetstvOS)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetsMacCatalyst)' == 'true'">
-      <MonoAOTCMakeArgs Include="-DCMAKE_OSX_DEPLOYMENT_TARGET=$(macOSVersionMin)" />
-    </ItemGroup>
 
+    <!-- Linux specific options -->
+    <ItemGroup Condition="$([MSBuild]::IsOSPlatform('Linux'))">
```
Comment (lambdageek) [inline src/mono/mono.proj:560]: Same - this is checking the machine that msbuild is running on. not the machine it's making binaries for - I think you want `RealTargetOS` if it is set.

**Review by lambdageek — COMMENTED** [review]
I'm not sure how much support we have for `build != host && host != target` kind of builds, but it seems like this is checking the machine that msbuild is running on, not the machine for which it is going to produce binaries.

File: src/mono/mono.proj (hunk 1 above)
Comment (akoeplinger) [reply src/mono/mono.proj:571]: Yeah, we're only building the AOT compiler on the same host OS as we'll be running on. I didn't know about `RealTargetOS`, will switch to that.

File: src/mono/mono.proj (hunk 2 above)
Comment (akoeplinger) [reply src/mono/mono.proj:560]: Yes, this was preexisting code that I just moved, will fix it too.

**Review by lambdageek — APPROVED** [review]

**Review by vargaz — APPROVED** [review]

**Review by vargaz — APPROVED** [review]

**Review by lewing — APPROVED** [review]

**Comment by lewing** [comment]
runtime (Mono crossaot Product Build Linux x64 release)

https://dev.azure.com/dnceng/public/_build/results?buildId=1296047&view=logs&j=5adba274-5943-551b-49fc-48c4614e2c6e&t=c1e3516a-fd93-5a03-0498-4dc70a478052&l=98

is https://github.com/dotnet/core-eng/issues/13691

**Comment by akoeplinger** [comment]
Failures are unrelated.

**Comment by lewing** [comment]
browser failures are https://github.com/dotnet/runtime/pull/57501

**Pull request merged**